### ISSUE TITLE
Update netty to 4.1.67

### ISSUE THREAD

**Issue by ThoSap** [grpc/grpc-java#8416]
### What version of gRPC-Java are you using?
https://github.com/grpc/grpc-java/releases/tag/v1.40.0

### What is your environment?

Linux & Windows

NexusIQ shows a vulnerability in netty-codec.
The version 4.1.66.Final and later has none.

Issue: SONATYPE-2021-0789
Fix part of Netty 4.1.66.Final: https://github.com/netty/netty/pull/11429

https://snyk.io/vuln/maven:io.netty%3Anetty-codec-http

**Comment by ThoSap** [comment]
https://github.com/quarkusio/quarkus/issues/18962

**Comment by ThoSap** [comment]
https://github.com/quarkusio/quarkus/pull/19426

**Comment by ThoSap** [comment]
**Netty milestones**

https://github.com/netty/netty/milestone/242 -> 4.1.67.Final
https://github.com/netty/netty/milestone/241 -> 4.1.66.Final
https://github.com/netty/netty/milestone/240 -> 4.1.65.Final
https://github.com/netty/netty/milestone/239 -> 4.1.64.Final
https://github.com/netty/netty/milestone/238 -> 4.1.63.Final
https://github.com/netty/netty/milestone/237 -> 4.1.62.Final
https://github.com/netty/netty/milestone/236 -> 4.1.61.Final
https://github.com/netty/netty/milestone/235 -> 4.1.60.Final
https://github.com/netty/netty/milestone/234 -> 4.1.59.Final
https://github.com/netty/netty/milestone/233 -> 4.1.58.Final
https://github.com/netty/netty/milestone/232 -> 4.1.57.Final
https://github.com/netty/netty/milestone/231 -> 4.1.56.Final
https://github.com/netty/netty/milestone/230 -> 4.1.55.Final
https://github.com/netty/netty/milestone/229 -> 4.1.54.Final
https://github.com/netty/netty/milestone/228 -> 4.1.53.Final

**Comment by ThoSap** [comment]
The current Netty version in use _4.1.52.Final_ (Netty v1.40.0) has multiple vulnerabilities in addition to the one mentioned above:
https://deps.dev/maven/io.netty%3Anetty-codec-http/4.1.52.Final

**Comment by ejona86** [comment]
gRPC is not impacted by the Gzip, Lz4, multipart vulnerabilities; gRPC does not use those classes.

> The only thing holding the Quarkus stack back to update to Netty 4.1.66.Final+ is this issue.

How are we holding up anything? You are free to upgrade the Netty version if it works, although then you have to debug issues more. Is the problem that there is an incompatibility between the newer Netty and gRPC? Based on the comments on the linked issue, it sounds like you suspect Netty 4.1.66 may have broke gRPC?

The quarkus-bot error shows something broke on server-side. Are the server-side logs available? Because they would be the ones that have the appropriate information to track down the problem.

I just tried running with Netty 4.1.66 and Netty-tcnative 2.0.40 and almost all grpc tests passed. One TLS test is hanging and it isn't clear to me why, but it started hanging with with Netty 4.1.64/65 so seems unrelated to what you are seeing.

**Comment by steveniemitz** [comment]
I stumbled over this because we're trying to upgrade to netty 4.1.67 but it seems to have broken gRPC.  I get this error:
```
Caused by: java.lang.UnsupportedOperationException: null
	at io.grpc.netty.AbstractHttp2Headers.setLong(AbstractHttp2Headers.java:465)
	at io.grpc.netty.AbstractHttp2Headers.setLong(AbstractHttp2Headers.java:26)
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:401)
```

it looks like the call to setLong was added in [4.1.60](https://github.com/netty/netty/blob/netty-4.1.60.Final/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java#L403) but is [unimplemented](https://github.com/grpc/grpc-java/blob/v1.40.0/netty/src/main/java/io/grpc/netty/AbstractHttp2Headers.java#L465) in grpc-netty.

**Comment by ejona86** [comment]
@steveniemitz, that issue (#7953) was resolved in v1.37.0.

**Comment by steveniemitz** [comment]
oh awesome, I missed looking at changes there when I was looking at changes.  Thanks! We'll bump gRPC too.

**Comment by ejona86** [comment]
@ThoSap, it appears your issue was resolved by vert-x3/vertx-grpc#106, so closing.

I'll note that we _do_ really want to upgrade Netty, and are working on it, but we've had limited time and there are issues to work through when upgrading (like a one-off testing framework incompatibility; nothing that others care about).

**Comment by ThoSap** [comment]
@ejona86 no worries. We will be patient and thank you for your awesome work! 😋
Testing is important to deliver quality software!
